test/sanity/run-test-v2.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
```

Comment:
`set -e` cause the script to immediately exit on an non-zero return from a command. This means that an intermediate failure in the `cleanup` function will leave artifacts behind. If we fail to cache the image in minikube, for example, `helm uninstall` will fail and prevent the rest of the clean up from occurring. You can either call `set +e` in the cleanup function, or append `|| true` to each command in `cleanup` to squelch errors (not necessary for `echo`, though).